src/Data/Nat/Primality/Rough.agda

```diff
@@ -0,0 +1,56 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Natural numbers whose prime factors are all above a bound
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Data.Nat.Primality.Rough where
+
+open import Data.Nat.Base using (ℕ; suc; _≤_; _<_; z≤n; s≤s; _+_)
+open import Data.Nat.Divisibility using (_∣_; _∤_; ∣-trans; ∣1⇒≡1)
+open import Data.Nat.Induction using (<-rec; <-Rec)
+open import Data.Nat.Primality using (Prime; composite?)
+open import Data.Nat.Properties using (_≟_; <-trans; ≤⇒≯; ≤∧≢⇒<; m<1+n⇒m<n∨m≡n)
+open import Data.Product.Base using (_,_)
+open import Data.Sum.Base using (inj₁; inj₂)
+open import Function.Base using (_∘_; flip)
+open import Relation.Nullary.Decidable.Core using (yes; no)
+open import Relation.Binary.PropositionalEquality.Core using (refl)
+
+private
+  variable
+    k m n : ℕ
+
+-- A number is k-rough if all its prime factors are greater than or equal to k
+_Rough_ : ℕ → ℕ → Set
+k Rough n = ∀ {d} → 1 < d → d < k → d ∤ n
+
+-- any number is 2-rough because all primes are greater than or equal to 2
+2-rough : 2 Rough n
+2-rough 1<d 2>d with () ← ≤⇒≯ 1<d 2>d
+
+extend-∤ : k Rough n → k ∤ n → suc k Rough n
```

Comment:
Can this lemma (since #2181 as `Data.Nat.Primality.∤⇒rough-suc`) be sharpened so that for `k > 2` it is possible to count up in increments of `2+ ` rather than merely `suc `?